src/Microsoft.TestPlatform.ObjectModel/TestProcessAttachDebuggerPayload.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.ObjectModel
+{
+    using System.Runtime.Serialization;
+
+    /// <summary>
+    /// The test process info payload.
+    /// </summary>
+    [DataContract]
+    public class TestProcessAttachDebuggerPayload
+    {
+        /// <summary>
+        /// Creates a new instance of this class.
+        /// </summary>
+        /// <param name="pid">The process id the debugger should attach to.</param>
+        public TestProcessAttachDebuggerPayload(int pid)
```

Comment:
>int pid [](start = 48, length = 7)

Well the other comment got me asking this - What is our plan for attaching to a process on another compute resource(docker container/remote VM)? Would we need to add anything else in this structure to avoid creating a TestProcessAttachDebuggerPayload2?

Reply:
I believe so far the remote debugger does supports attaching to process by PID in the docker case and that would not require any API changes here. There is a bug with the old launch and attach workflow where the debugger APIs were not returning the PID - @drognanar is working with the docker / debugger folks on a fix for that.

Reply:
I see, would a property bag still be useful in this case? In case Attach needs additional information.

Reply:
Do you mean an open-eded dictionary of string-object or something else? I'd hate to add something to the API without understanding the scenario. Its probably ok to rev the API again if we need to - but I don't think we should create an API without fully understanding how it will be used.

Reply:
So one of the things that came up for the LaunchProcessWithDebuggerAttached was to be able to pass back DebuggerGuids. With Attach though, I'm thinking off something like remote machine details for instance. A string-string would just do given the content can be serialized.

Reply:
I think it would be better to wait until those needs become clear before we change the API. @drognanar Do you know of any changes that are absolutely required on the TP-side for the Docker changes?

Reply:
Right now I'm definitely taking a dependency on us using vsdbg to start a debugger and passing a JSON specifically for .NET. My original thought was to link the debugger choice back to the containers. If we would want to have a C# project launch a debugger for a node.exe, then we would need to have additional parameters in this method.

Also, I've been passing back RemoteEnvironment from the request back to VS, so we don't need to specify that again in the testhost (unless the testhost would start a test process on another machine than where the testhost lives).

Reply:
In short currently we're using the information from the test container, the runtime provider and the pid in order to create an appropriate attach request.

Reply:
Yeah, non-managed workflows was what I was thinking off as well. I'm not sure what other parameters would be required there.

Reply:
Adding in a string-string property bag and terming it as options wouldn't hurt much I'd think.

Reply:
I think if we have to do that we'd also need to add it to the LaunchAndAttach API for completeness - including to the IFrameworkHandle APIs that are exposed to adapter.

I'm a bit loath to do all that as part of the current change as its not required for this change + we haven't really proved that we need it (as far as I can tell we don't have a real need for it for Docker so far). We may discover that we need something else / different API shape when we actually get to implementing remote debugging support where additional data is required (for example SSH / remote VM).

So I feel it may be best to wait and introduce those APIs as and when the need arises.